### PR TITLE
Add recalcOrderPercentageJob, integrationTests

### PR DESCRIPTION
**Pull request by badekk**
## Add recalcOrderPercentageJob, integrationTests

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How has this been tested?

- [ ] Unit test(s)
- [x] Integration test(s)
- [ ] Tested manually

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I have checked my code and corrected any misspellings
- [ ] I have added frontend code and checked it for mobile & desktop

### CODE REVIEW

**Comment by mar-pan** [comment]
Task linked: [CU-862jfe3by RecalcOrdersPercentageJob](https://app.clickup.com/t/862jfe3by)

File: src/backend/Core/Migrations/AppDbContextModelSnapshot.cs [hunk 1]
```diff
@@ -3408,6 +3408,10 @@ protected override void BuildModel(ModelBuilder modelBuilder)
                         .HasColumnType("boolean")
                         .HasColumnName("is_canceled_by_user");
 
+                    b.Property<bool>("IsFinished")
+                        .HasColumnType("boolean")
```
Comment (h4570) [inline src/backend/Core/Migrations/AppDbContextModelSnapshot.cs:3412]: Is required, bo nie masz `bool?`

File: src/backend/Orders/Jobs/RecalcOrdersPercentage/RecalcOrdersPercentageJob.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            _recalcOrdersPercentageJobDataService = recalcOrdersPercentageJobDataService;
+            _recalcOrdersPercentageJobService = recalcOrdersPercentageJobService;
+            _appDbContext = appDbContext;
+        }
+
+        public async Task<ActionResult> Execute()
+        {
+            var listOfOrders = await _recalcOrdersPercentageJobDataService.Fetch();
+
+            if (listOfOrders != null)
```
Comment (h4570) [inline src/backend/Orders/Jobs/RecalcOrdersPercentage/RecalcOrdersPercentageJob.cs:26]: To nigdy nie będzie prawdą. Wywal

File: src/backend/Orders/Jobs/RecalcOrdersPercentage/RecalcOrdersPercentageJobService.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            int percentage = (int)Math.Round((double)(deliveredQuantity * 100) / totalQuantity);
+
+            if (percentage >= 100)
+            {
+                order.IsFinished = true;
+            }
+
+            order.Percentage = percentage;
+
+            await _appDbContext.SaveChangesAsync();
```
Comment (h4570) [inline src/backend/Orders/Jobs/RecalcOrdersPercentage/RecalcOrdersPercentageJobService.cs:32]: Po co robić strzał do bazy co iterację. Przecież masz tam po pętli zapis. Wywal

**Review by h4570 — CHANGES_REQUESTED** [review]

File: src/backend/Core/Migrations/AppDbContextModelSnapshot.cs (hunk 1 above)
Comment (badekk) [reply src/backend/Core/Migrations/AppDbContextModelSnapshot.cs:3412]: Bo w configu dałem 
            builder.Property(c => c.IsFinished).IsRequired();
ustawiłem tak jak inne były ustawione

File: src/backend/Orders/Jobs/RecalcOrdersPercentage/RecalcOrdersPercentageJob.cs (hunk 2 above)
Comment (badekk) [reply src/backend/Orders/Jobs/RecalcOrdersPercentage/RecalcOrdersPercentageJob.cs:26]: w sumie to powinno być listOfOrders.Any()... 
ale masz rację, mimo wszystko foreach nie wykona się gdy lista jest pusta.

File: src/backend/Orders/Jobs/RecalcOrdersPercentage/RecalcOrdersPercentageJobService.cs (hunk 3 above)
Comment (badekk) [reply src/backend/Orders/Jobs/RecalcOrdersPercentage/RecalcOrdersPercentageJobService.cs:32]: true

**Pull request merged**